src/test/java/org/folio/rest/impl/NoteTypesImplTest.java

```diff
@@ -58,6 +62,169 @@ public void shouldReturn200WithNoteTypeWhenValidId() throws IOException, URISynt
+    try {
+      final String stubNoteType = readFile(POST_NOTE_TYPE_JSON);
+
+      NoteTypesTestUtil.insertNoteType(vertx, ID, stubNoteType);
+
+      Response response = RestAssured.given()
+        .spec(getRequestSpecification())
+        .when()
+        .get(NOTE_TYPES_ENDPOINT)
+        .then()
+        .statusCode(200)
+        .extract().response();
+
+      int totalRecords = response.path(TOTAL_RECORDS);
+      List<NoteType> noteTypes = response.path(NOTE_TYPES);
+
+      assertEquals(1, noteTypes.size());
+      assertEquals(1, totalRecords);
+    } finally {
+      NoteTypesTestUtil.deleteAllNoteTypes(vertx);
+    }
+  }
+
+  @Test
+  public void shouldReturn200WithNoteTypeCollectionAndWithQuery() throws IOException, URISyntaxException {
+    try {
+      ObjectMapper mapper = new ObjectMapper();
+      NoteType[] noteTypes = mapper.readValue(readFile(COLLECTION_NOTE_TYPE_JSON), NoteType[].class);
+
+      for (NoteType noteType : noteTypes) {
+        NoteTypesTestUtil.insertNoteType(vertx, noteType.getId(), mapper.writeValueAsString(noteType));
+      }
+      Response response = RestAssured.given()
+        .spec(getRequestSpecification())
+        .when()
+        .get(NOTE_TYPES_ENDPOINT + "?query=(name=High)")
+        .then()
+        .statusCode(200)
+        .extract().response();
+
+      int totalRecords = response.path(TOTAL_RECORDS);
+      List<NoteType> noteTypeList = response.path(NOTE_TYPES);
+
+      assertEquals(1, noteTypeList.size());
+      assertEquals(1, totalRecords);
+    } finally {
+      NoteTypesTestUtil.deleteAllNoteTypes(vertx);
+    }
+  }
+
+  @Test
+  public void shouldReturn200WithThreeNoteTypeElements() throws IOException, URISyntaxException {
+    try {
+      ObjectMapper mapper = new ObjectMapper();
+      NoteType[] noteTypes = mapper.readValue(readFile(COLLECTION_NOTE_TYPE_JSON), NoteType[].class);
+
+      for (NoteType noteType : noteTypes) {
+        NoteTypesTestUtil.insertNoteType(vertx, noteType.getId(), mapper.writeValueAsString(noteType));
+      }
+      Response response = RestAssured.given()
+        .spec(getRequestSpecification())
+        .when()
+        .get(NOTE_TYPES_ENDPOINT)
+        .then()
+        .statusCode(200)
+        .extract().response();
+
+      int totalRecords = response.path(TOTAL_RECORDS);
+      List<NoteType> noteTypeList = response.path(NOTE_TYPES);
+
+      assertEquals(3, noteTypeList.size());
+      assertEquals(3, totalRecords);
+    } finally {
+      NoteTypesTestUtil.deleteAllNoteTypes(vertx);
+    }
+  }
+
+  @Test
+  public void shouldReturn200WithNoteTypeCollectionAndIncompleteWay() throws IOException, URISyntaxException {
+    try {
+      final String stubNoteType = readFile(POST_NOTE_TYPE_JSON);
+
+      NoteTypesTestUtil.insertNoteType(vertx, ID, stubNoteType);
+
+      Response response = RestAssured.given()
+        .spec(getRequestSpecification())
+        .when()
+        .get(NOTE_TYPES_ENDPOINT + "?quer")
+        .then()
```

Comment:
Shouldn't this give a bad request or is the parameter simply ignored?

Reply:
Thanks Carole --If request has wrong parameter(like typo or incomplete) than it would be ignored